### PR TITLE
Remove deprecated codes for release 2.8

### LINKED ISSUE #4766: Remove features deprecated in 2.6 and slated to be removed in 2.8
There's a handful of configuration options or code deprecated in Mimir 2.6 that need to be removed in Mimir 2.8. Tracking issue for that work.

* From [`pkg/ingester/shipper.go`](https://github.com/grafana/mimir/blob/b3af8077c7e91e8b1ccb5e678be8cb710907b819/pkg/ingester/shipper.go#L265-L282).

* From [`pkg/util/validation/limits.go`](https://github.com/grafana/mimir/blob/b3af8077c7e91e8b1ccb5e678be8cb710907b819/pkg/util/validation/limits.go#L119) - remove `max_query_length`.

### PR DESCRIPTION
**Pull request by lamida**
#### Which issue(s) this PR fixes or relates to

Fixes #4766

#### Checklist

- NA Tests updated
- NA Documentation added
- NA `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pracucci — APPROVED** [review]
LGTM, but please also remove the `-store.max-query-length` reference from `docs/sources/mimir/operators-guide/mimir-runbooks/_index.md`.

File: pkg/ingester/shipper.go [hunk 1]
```diff
@@ -252,7 +252,12 @@ func (s *Shipper) blockMetasFromOldest() (metas []*metadata.Meta, _ error) {
 func readShippedBlocks(dir string) (map[ulid.ULID]time.Time, error) {
 	meta, err := readShipperMetaFile(dir)
 	if err != nil {
-		return readThanosShippedBlocks(dir)
+		if errors.Is(err, os.ErrNotExist) {
+			// If the meta file doesn't exist it means the shipper hasn't run yet.
+			meta = shipperMeta{}
+		} else if err != nil {
```
Comment (krajorama) [inline pkg/ingester/shipper.go:258]:
```suggestion
		} else {
```

File: pkg/ingester/shipper.go [hunk 2]
```diff
@@ -252,7 +252,12 @@ func (s *Shipper) blockMetasFromOldest() (metas []*metadata.Meta, _ error) {
 func readShippedBlocks(dir string) (map[ulid.ULID]time.Time, error) {
 	meta, err := readShipperMetaFile(dir)
 	if err != nil {
```
Comment (56quarters) [inline pkg/ingester/shipper.go:254]: nit but, this whole block could be simplified a bit I think:
```go
if err != nil && errors.Is(err, os.ErrNotExist) {
    meta = shipperMeta{}
} else if err != nil {
    return nil, err
}
```

File: pkg/util/validation/limits.go [hunk 3]
```diff
@@ -222,8 +221,6 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&l.MaxChunksPerQuery, MaxChunksPerQueryFlag, 2e6, "Maximum number of chunks that can be fetched in a single query from ingesters and long-term storage. This limit is enforced in the querier, ruler and store-gateway. 0 to disable.")
 	f.IntVar(&l.MaxFetchedSeriesPerQuery, MaxSeriesPerQueryFlag, 0, "The maximum number of unique series for which a query can fetch samples from each ingesters and storage. This limit is enforced in the querier, ruler and store-gateway. 0 to disable")
 	f.IntVar(&l.MaxFetchedChunkBytesPerQuery, MaxChunkBytesPerQueryFlag, 0, "The maximum size of all chunks in bytes that a query can fetch from each ingester and storage. This limit is enforced in the querier and ruler. 0 to disable.")
-	// TODO: Deprecated in Mimir 2.6, remove in Mimir 2.8
-	f.Var(&l.MaxQueryLength, maxQueryLengthFlag, fmt.Sprintf("Deprecated: Limit the query time range (end - start time). This limit is enforced in the querier (on the query possibly split by the query-frontend) and ruler. 0 to disable. This option is deprecated, use -%s or -%s instead.", maxPartialQueryLengthFlag, maxTotalQueryLengthFlag))
```
Comment (56quarters) [inline pkg/util/validation/limits.go:226]: `maxQueryLengthFlag` should probably be removed too, right?

File: pkg/util/validation/limits.go [hunk 4]
```diff
@@ -222,8 +221,6 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
...
 	f.Var(&l.MaxPartialQueryLength, maxPartialQueryLengthFlag, fmt.Sprintf("Limit the time range for partial queries at the querier level. Defaults to the value of -%s if set to 0.", maxQueryLengthFlag))
```
Comment (56quarters) [inline pkg/util/validation/limits.go:224]: The help here and for the total length setting references `store.max-query-length` and needs to be updated.

**Comment by pstibrany** [comment]
When we target `main` branch, steps are: 1. merge to `main`, 2. cherry-pick to `release-2.8`, 3. merge release branch to `main`.

If we target `release-2.8` branch with this PR, we would save step 2, and possibly one conflict resolution. End result would be the same.

File: pkg/util/validation/limits.go (hunk 3 above)
Comment (lamida) [reply pkg/util/validation/limits.go:226]: I was searching "2.8" yesterday, and couldn't find anything else. Weird. Thanks for catching this up.

**Comment by lamida** [comment]
> When we target main branch, steps are: 1. merge to main, 2. cherry-pick to release-2.8, 3. merge release branch to main.

Yes realised after making some changes and too lazy to change the target branch to release-2.8. I will update the documentation for this step too. For now I will just making sure following the cherry-pick step.

File: pkg/ingester/shipper.go (hunk 2 above)
Comment (lamida) [reply pkg/ingester/shipper.go:254]: I am taking @krajorama suggestion which looks more cleaner.

```
if err != nil {
		if errors.Is(err, os.ErrNotExist) {
			// If the meta file doesn't exist it means the shipper hasn't run yet.
			meta = shipperMeta{}
		} else  {
			return nil, err
		}
	}
```

**Review by krajorama — APPROVED** [review]

**Pull request merged**